tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -95,64 +106,72 @@ public void Authentication_succeeds_when_user_has_both_Scram_Sha_mechanisms_and_
             var settings = client.Settings.Clone();
             settings.Credential = MongoCredential
                 .FromComponents(mechanism: null, source: null, username: userName, password: password);
-            
-            AssertAuthenticationSucceeds(settings);
+
+            AssertAuthenticationSucceeds(settings, async);
         }
-        
-        [SkippableFact]
-        public void Authentication_succeeds_when_user_has_both_scram_sha_mechanisms_and_mechanism_is_Scram_Sha_256()
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Authentication_succeeds_when_user_has_both_scram_sha_mechanisms_and_mechanism_is_Scram_Sha_256(
+            [Values(false, true)] bool async)
         {
             RequireServer.Check().Supports(Feature.ScramSha256Authentication).Authentication(true);
-            
+
             var client = DriverTestConfiguration.Client;
             var userName = $"both{Guid.NewGuid()}";
             var password = "both";
             CreateAdminDatabaseReadWriteUser(client, userName, password, "SCRAM-SHA-256", "SCRAM-SHA-1");
             var settings = client.Settings.Clone();
             settings.Credential = MongoCredential
                 .FromComponents(mechanism: "SCRAM-SHA-256", source: null, username: userName, password: password);
-            
-            AssertAuthenticationSucceeds(settings);
+
+            AssertAuthenticationSucceeds(settings, async);
         }
-        
-        [SkippableFact]
-        public void Authentication_succeeds_when_user_has_Scram_Sha_1_Mechanism_and_mechanism_is_not_specified()
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Authentication_succeeds_when_user_has_Scram_Sha_1_Mechanism_and_mechanism_is_not_specified(
+            [Values(false, true)] bool async)
         {
             RequireServer.Check().Supports(Feature.ScramSha256Authentication).Authentication(true);
             // mechanisms field in createUser command requires server >=4.0
             var client = DriverTestConfiguration.Client;
-            
+
             var userName = $"sha1{Guid.NewGuid()}";
             var password = "sha1";
             CreateAdminDatabaseReadWriteUser(client, userName, password, "SCRAM-SHA-1");
             var settings = client.Settings.Clone();
             settings.Credential = MongoCredential
                 .FromComponents(mechanism: null, source: null, username: userName, password: password);
 
-            AssertAuthenticationSucceeds(settings);
+            AssertAuthenticationSucceeds(settings, async);
         }
-        
-        [SkippableFact]
```

Comment:
Forgetting to change the Facts to Theories yields a somewhat obscure message that took me a while to figure out:
`  System.InvalidOperationException : The test method expected 1 parameter value, but 0 parameter values were provided.`